scripts/linters/pylint_extensions.py

```diff
@@ -385,6 +407,102 @@ def check_functiondef_params(self, node, node_doc):
+                    if current_line_indentation % 4 != 0:
+                        self.add_message(
+                            'incorrect-indentation-for-arg-header-doc',
+                            node=node)
+                        #Since other checks are based on relative indentation,
+                        #we need to fix this indentation first.
+                        break
+                    currently_in_args_section = True
+                # Check for parameter section header by checking that the
+                # parameter is in the function arguments set. We also check for
+                # arguments that start with * which means it's autofill and will
+                # not appear in the node args list so we handle those too.
+                elif (currently_in_args_section and parameter
+                      and ((parameter.group(0).strip('*')
+                            in expected_argument_names) or
+                           re.search(
+                               br'\*[^ ]+: ',
+                               line_with_no_leading_whitespaces))):
+                    currently_in_freeform_section = False
+                    # Check if the current parameter section indentation is
+                    # correct.
+                    if current_line_indentation != (
+                            args_indentation + 4):
+                        # Use the first word in the line to identify the error.
+                        beginning_of_line = (
+                            (None) if
+                            (not line_with_no_leading_whitespaces.split(' '))
+                            else line_with_no_leading_whitespaces.split(' ')[0])
+                        self.add_message(
+                            '4-space-indentation-for-arg-parameters-doc',
+                            node=node,
+                            args=(beginning_of_line))
+                    # If the line ends with a colon, that means
+                    # the next subsection of of in_description is free form.
+                    if line.endswith(':'):
+                        currently_in_freeform_section = True
+                # If we get to a different section, we exit the loop.
+                elif line.strip() in self.docstring_sections:
+                    break
+                # All other lines can be treated as in_description.
+                elif currently_in_args_section:
+                    # If it is not a freeform section, we check the indentation.
+                    if (not currently_in_freeform_section
+                            and current_line_indentation != (
+                                args_indentation + 8)):
+                        # Use the first word in the line to identify the error.
+                        beginning_of_line = (
+                            (None) if
+                            (not line_with_no_leading_whitespaces.split(' '))
```

Comment:
Same note as above.

Also, I think you don't need most of the parens here -- at least around None, or around beginning_of_line below.

Reply:
Done